ui/src/components/constants.ts

```diff
@@ -54,6 +54,7 @@ export const STEP_LABELS: Record<PipelineStepName, string> = {
   todate: 'Convert text to date',
   fromdate: 'Convert date to text',
   dateextract: 'Extract date information',
+  dategranularity: 'Apply date granularity',
```

Comment:
```suggestion
  dategranularity: 'Normalize to date granularity',
```

proposal: I think this is clearer, as to me,  "apply date granularity" doesn't really describe what the step does